internal/util/auxutil.go

```diff
@@ -0,0 +1,88 @@
+package util
+
+import (
+	"slices"
+
+	etcdaenixiov1alpha1 "github.com/aenix-io/etcd-operator/api/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EtcdClusterIsReady returns true is condition "Ready" has status equals to "True", otherwise false.
+func EtcdClusterIsReady(cluster *etcdaenixiov1alpha1.EtcdCluster) bool {
+	idx := slices.IndexFunc(cluster.Status.Conditions, func(condition metav1.Condition) bool {
+		return condition.Type == etcdaenixiov1alpha1.EtcdConditionReady
+	})
+	if idx == -1 {
+		return false
+	}
+	return cluster.Status.Conditions[idx].Status == metav1.ConditionTrue
+}
+
+// FillConditions fills EtcdCluster .status.Conditions list with all available conditions in "False" status.
+func FillConditions(cluster *etcdaenixiov1alpha1.EtcdCluster) {
+	conditions := []string{
+		etcdaenixiov1alpha1.EtcdConditionInitialized,
+		etcdaenixiov1alpha1.EtcdConditionReady,
+	}
+
+	for _, c := range conditions {
+		SetCondition(cluster, c, false)
+	}
+}
+
+// SetCondition sets the condition of defined type to desired state.
+func SetCondition(
+	cluster *etcdaenixiov1alpha1.EtcdCluster,
+	conditionType string,
```

Comment:
I would use `etcdaenixiov1alpha1.EtcdCondType` here

Reply:
SetCondition function updated as agreed in the comments thread below